Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewRow.cs

```diff
@@ -0,0 +1,710 @@
+            }
+        }
+
+        // <summary>
+        // Gets a value indicating whether the row is the row for new records.
+        // </summary>
+        // 
+        // <returns>
+        // true if the row is the last row in the <see cref="T:System.Windows.Forms.DataGridView"/>, which is used for the entry of a new row of data; otherwise, false.
+        // </returns>
+        // public bool IsNewRow {get;}
+        
+        /// <summary>
+        /// Gets or sets the minimum height of the row.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The minimum row height in pixels, ranging from 2 to <see cref="F:System.Int32.MaxValue"/>. The default is 3.
+        /// </returns>
+        /// <exception cref="T:System.InvalidOperationException">When setting this property, the row is in a <see cref="T:System.Windows.Forms.DataGridView"/> control and is a shared row.</exception><exception cref="T:System.ArgumentOutOfRangeException">The specified value when setting this property is less than 2.</exception><PermissionSet><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+        public int MinimumHeight
+        {
+            get { return default(int); }
+            set { Contract.Requires(!(this.DataGridView != null && this.Index == -1)); }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the row is read-only.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// true if the row is read-only; otherwise, false.
+        /// </returns>
+        /// <exception cref="T:System.InvalidOperationException">The row is in a <see cref="T:System.Windows.Forms.DataGridView"/> control and is a shared row.</exception><PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.SecurityPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Flags="UnmanagedCode, ControlEvidence"/><IPermission class="System.Diagnostics.PerformanceCounterPermission, System, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+        [DefaultValue(false)]
+        public bool ReadOnly
+        {
+            get
+            {
+                Contract.Requires(!(this.DataGridView != null && this.Index == -1));
+                return default(bool);
+            }
+            set
+            {
+                
+            }
+        }
+
```

Comment:
The same issue with regular pattern for Index != -1 check for other properties.....